control-plane/pkg/apis/eventing/v1alpha1/kafka_sink_types.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+const (
+	// CloudEvents binary content mode.
+	ModeBinary = "binary"
+	// CloudEvents structured content mode.
+	ModeStructured = "structured"
+)
+
+var allowedContentModes = sets.NewString(ModeStructured, ModeBinary)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KafkaSink is an addressable resource that represent a Kafka topic.
+type KafkaSink struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the Kafka Sink.
+	Spec KafkaSinkSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the KafkaSink.
+	// This data may be out of date.
+	// +optional
+	Status KafkaSinkStatus `json:"status,omitempty"`
+}
+
+// Check that Channel can be validated, can be defaulted, and has immutable fields.
+var _ apis.Validatable = (*KafkaSink)(nil)
+var _ apis.Defaultable = (*KafkaSink)(nil)
+var _ runtime.Object = (*KafkaSink)(nil)
+var _ duckv1.KRShaped = (*KafkaSink)(nil)
+
+// KafkaSinkSpec defines the desired state of the Kafka Sink.
+type KafkaSinkSpec struct {
+	Topic string `json:"topic"`
+
+	NumPartitions int32 `json:"numPartitions"`
```

Comment:
wait are we going to create a new topic? i assume the point of the sink was to push stuff to existing topics... I wonder, should we support both?

Reply:
If the topic is already there we simply don't create it.

Reply:
in that use case, all the params except topic doesn't make sense here... Maybe they should be optional?

Reply:
They are optional, https://github.com/knative-sandbox/eventing-kafka-broker/pull/134/files#diff-8dc0776390d25e25ee94315d523384b3R41, the `+optional` is a maker used by `kubebuilder` that we don't use, I'll add it.